File: drivelite-backend/drivelite-backend/src/main/java/com/drivelite/service/VehicleService.java
package com.drivelite.service;

import com.drivelite.model.Vehicle;
import com.drivelite.repository.VehicleRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class VehicleService {

    private final VehicleRepository vehicleRepository;

    public VehicleService(VehicleRepository vehicleRepository) {
        this.vehicleRepository = vehicleRepository;
    }

    // Add Vehicle
    public Vehicle createVehicle(Vehicle vehicle) {
        return vehicleRepository.save(vehicle);
    }

    // Get All Vehicles
    public List<Vehicle> getAllVehicles() {
        return vehicleRepository.findAll();
    }

    // Get Vehicle By ID
    public Vehicle getVehicleById(Integer id) {

        return vehicleRepository.findById(id)
                .orElseThrow(() ->
                        new RuntimeException("Vehicle not found"));
    }

    // Update Vehicle
    public Vehicle updateVehicle(Integer id, Vehicle vehicle) {

        Vehicle existingVehicle = getVehicleById(id);

        existingVehicle.setVehicleName(vehicle.getVehicleName());
        existingVehicle.setVehicleType(vehicle.getVehicleType());
        existingVehicle.setModel(vehicle.getModel());
        existingVehicle.setNumberPlate(vehicle.getNumberPlate());
        existingVehicle.setPricePerDay(vehicle.getPricePerDay());
        existingVehicle.setAvailable(vehicle.isAvailable());

        return vehicleRepository.save(existingVehicle);
    }

    // Delete Vehicle
    public String deleteVehicle(Integer id) {

        vehicleRepository.deleteById(id);

        return "Vehicle deleted successfully";
    }

    // Filter by type
    public List<Vehicle> getVehiclesByType(String type) {
        return vehicleRepository.findByVehicleType(type);
    }

    // Available vehicles
    public List<Vehicle> getAvailableVehicles() {
        return vehicleRepository.findByAvailableTrue();
    }

    // Price range
    public List<Vehicle> getVehiclesByPrice(double min, double max) {
        return vehicleRepository.findByPricePerDayBetween(min, max);
    }
}

File: drivelite-frontend/src/services/vehicleService.js
import api from "./api";


// ================= USER VEHICLE APIs =================


// Get all vehicles
export const getVehicles = () => {
    return api.get("/vehicles");
};


// Get available vehicles
export const getAvailableVehicles = () => {
    return api.get("/vehicles/available");
};


// Get vehicles by type
export const getVehicleByType = (type) => {
    return api.get(`/vehicles/type/${type}`);
};


// Get vehicles by price range
export const getVehiclesByPrice = (min, max) => {
    return api.get(`/vehicles/price?min=${min}&max=${max}`);
};



// ================= ADMIN VEHICLE CRUD APIs =================


// Add vehicle
export const createVehicle = (vehicle) => {
    return api.post("/vehicles", vehicle);
};


// Update vehicle
export const updateVehicle = (id, vehicle) => {
    return api.put(`/vehicles/${id}`, vehicle);
};


// Delete vehicle
export const deleteVehicle = (id) => {
    return api.delete(`/vehicles/${id}`);
};